_posts/2019-01-03-elixir-kaffe-codealong.md

```diff
@@ -0,0 +1,204 @@
+# Connecting Elixir to Kafka with Kaffe: A codealong
+_The most basic example that exists_
+
+If we want to use the popular messaging system Kafka with our Elixir projects, we have a few wrappers we can choose from. One of them, [Kaffe](https://github.com/spreedly/kaffe), proved difficult to troubleshoot. Just try googling "Elixir Kaffe" yourself.
+
+In this codealong we'll build the barest of simple Elixir applications and use Kaffe to connect it to a locally running Kafka server. Later we'll cover a couple of variations to connect a dockerized Kafka server or an umbrella elixir app.
+
+This post assumes basic knowledge of Elixir and no knowledge of Kafka or Kaffe. Here is the repo with the full project: [Elixir Kaffe Codealong](https://github.com/meryldakin/elixir_kaffe_codealong).
+
+## What is Kafka, briefly?
+Kafka is a messaging system. It does essentially three things:
+1. Receives messages from applications
+2. Keeps those messages in the order they were receieved in
+3. Allows other applications to read those messages in order
+
+Learn a lot more here: [Kafka in a Nutshell](https://sookocheff.com/post/kafka/kafka-in-a-nutshell/)
+
```

Comment:
This would be good to have. We don't need anything super detailed but some folks have never used Kafka or a message system and may not understand how it fits in.  The more we can explain here, without going off topic, and without sending them offsite, the better 👍